src/storage/single_file_block_manager.cpp

```diff
@@ -226,13 +226,17 @@ void SingleFileBlockManager::LoadExistingDatabase() {
 void SingleFileBlockManager::ReadAndChecksum(FileBuffer &block, uint64_t location) const {
 	// read the buffer from disk
 	block.Read(*handle, location);
+
 	// compute the checksum
 	auto stored_checksum = Load<uint64_t>(block.InternalBuffer());
 	uint64_t computed_checksum = Checksum(block.buffer, block.size);
+
 	// verify the checksum
 	if (stored_checksum != computed_checksum) {
-		throw IOException("Corrupt database file: computed checksum %llu does not match stored checksum %llu in block",
-		                  computed_checksum, stored_checksum);
+		throw IOException(
```

Comment:
Can we revert this error message, and instead add the blocksize of the database file to the `DatabaseHeader`? That way we can detect mismatched block sizes in an earlier stage.